### PR TITLE
Refactoring the Android_views tests app to prepare for adding the iOS platform view tests

### PR DESCRIPTION
**Pull request by cyanglaz**
## Description

This PR created a main page for platform view tests in the android views testing app. The main page none contains a list of the links to the pages being tested. It puts the android view motion events tests to a sub page.
The PR also added iOS related files to get ready for adding the iOS platform views tests.

Following up: 
1. A PR to rename the app to platform_views. 
2. A PR to add the platform views mutation screenshot tests.

## Tests

I added the following tests:

No new tested added, modified the `recomposition` tests with navigation taps.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flutter developers to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (Please read [Handling breaking changes]). *Replace this with a link to the e-mail where you asked for input on this proposed change.*
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: dev/integration_tests/android_views/lib/page.dart [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+
+abstract class Page extends StatelessWidget {
+  const Page(this.title, this.valueKey);
+
+  final String title;
+  final ValueKey<String> valueKey;
```
Comment (amirh) [inline dev/integration_tests/android_views/lib/page.dart]: Why do we need this?

File: dev/integration_tests/android_views/lib/page.dart [hunk 2]
```diff
@@ -0,0 +1,12 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+
+abstract class Page extends StatelessWidget {
```
Comment (amirh) [inline dev/integration_tests/android_views/lib/page.dart]: Document

File: dev/integration_tests/android_views/lib/page.dart (hunk 1 above)
Comment (amirh) [reply dev/integration_tests/android_views/lib/page.dart]: Coming back here after reading the test code now I see, document it though.

File: dev/integration_tests/android_views/test_driver/main_test.dart [hunk 3]
```diff
@@ -3,17 +3,31 @@
...
+  tearDownAll(() {
+    driver.close();
   });
-}
 
+  group('MotionEvents tests ', () {
+    test('recomposition', () async {
+      if (Platform.isAndroid) {
+        final SerializableFinder linkToOpenMotionEventPage =
+        find.byValueKey('MotionEventPage');
```
Comment (amirh) [inline dev/integration_tests/android_views/test_driver/main_test.dart]: Sounds like this should say `MotionEventsListTile`

File: dev/integration_tests/android_views/test_driver/main_test.dart [hunk 4]
```diff
@@ -3,17 +3,31 @@
...
+        await driver.tap(linkToOpenMotionEventPage);
+        await driver.waitFor(find.byValueKey('MotionEventPageLoaded'));
```
Comment (amirh) [inline dev/integration_tests/android_views/test_driver/main_test.dart]: Call this key after the element it's used for.

File: dev/integration_tests/android_views/lib/motion_event_page.dart [hunk 5]
```diff
@@ -0,0 +1,297 @@
...
+import 'package:path_provider/path_provider.dart';
+
+import 'motion_event_diff.dart';
+import 'page.dart';
+
+MethodChannel channel = const MethodChannel('android_views_integration');
+
+const String kEventsFileName = 'touchEvents';
+
+class MotionEventPage extends Page {
```
Comment (amirh) [inline dev/integration_tests/android_views/lib/motion_event_page.dart]: nit: MotionEventsPage

File: dev/integration_tests/android_views/lib/motion_event_page.dart [hunk 6]
```diff
@@ -0,0 +1,297 @@
...
+              onPressed: () {
+                const StandardMessageCodec codec = StandardMessageCodec();
+                saveRecordedEvents(
+                    codec.encodeMessage(flutterViewEvents), context);
+              },
+            ),
+            RaisedButton(
+              key: const ValueKey<String>('play'),
+              child: const Text('PLAY FILE'),
+              onPressed: () {
```
Comment (amirh) [inline dev/integration_tests/android_views/lib/motion_event_page.dart]: Avoid reformatting the previous code if the previous formatting is good. (in multiple places)

File: dev/integration_tests/android_views/lib/motion_event_diff.dart [hunk 7]
```diff
@@ -106,7 +106,8 @@ void diffPointerCoordsList(StringBuffer diffBuffer,
 
 void diffPointerCoords(Map<String, dynamic> expected,
     Map<String, dynamic> actual, int pointerIdx, StringBuffer diffBuffer) {
-  diffMaps(expected, actual, diffBuffer, messagePrefix: '[pointerCoord $pointerIdx] ');
+  diffMaps(expected, actual, diffBuffer,
```
Comment (amirh) [inline dev/integration_tests/android_views/lib/motion_event_diff.dart]: avoid unneeded reformatting

**Comment by cyanglaz** [comment]
@amirh Updated. Ready for another review.

File: dev/integration_tests/android_views/lib/motion_event_diff.dart [hunk 8]
```diff
@@ -123,9 +123,9 @@ void diffMaps(
     return;
   }
   for (String key in expected.keys) {
-    if (excludeKeys.contains(key))
+    if (excludeKeys.contains(key))
```
Comment (amirh) [inline dev/integration_tests/android_views/lib/motion_event_diff.dart]: nit: remove extra space

File: dev/integration_tests/android_views/lib/motion_event_page.dart [hunk 9]
```diff
@@ -0,0 +1,292 @@
...
+import 'motion_event_diff.dart';
+import 'page.dart';
+
+MethodChannel channel = const MethodChannel('android_views_integration');
+
+const String kEventsFileName = 'touchEvents';
+
+class MotionEventsPage extends Page {
+  const MotionEventsPage()
+      : super('Place circle', const ValueKey<String>('MotionEventsListTile'));
```
Comment (amirh) [inline dev/integration_tests/android_views/lib/motion_event_page.dart]: Place circle?

File: dev/integration_tests/android_views/lib/motion_event_diff.dart [hunk 10]
```diff
@@ -123,9 +123,9 @@ void diffMaps(
     return;
   }
   for (String key in expected.keys) {
-    if (excludeKeys.contains(key))
+    if (excludeKeys.contains(key)) 
       continue;
-    if (doublesApproximatelyMatch(expected[key], actual[key]))
+    if (doublesApproximatelyMatch(expected[key], actual[key]))
```
Comment (amirh) [inline dev/integration_tests/android_views/lib/motion_event_diff.dart]: here as well

**Review by amirh — APPROVED** [review]
LGTM module nits and renaming motion_event_page.dart to motion_events_page.dart

**Comment by cyanglaz** [comment]
Updated!

**Pull request merged**